iphone/Classes/TiApp.m

```diff
@@ -780,30 +780,61 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didSend
   [[NSNotificationCenter defaultCenter] postNotificationName:kTiURLUploadProgress object:self userInfo:dict];
 }
 
-- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
+-(void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveData:(NSData *)data
 {
-  NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithObjectsAndKeys:
-                                                       [NSNumber numberWithUnsignedInteger:task.taskIdentifier], @"taskIdentifier",
-                                                   nil];
+    if(!uploadTaskResponses){
+        uploadTaskResponses = [[NSMutableDictionary alloc] init];
+    }
+    NSMutableDictionary *responseObj =  [uploadTaskResponses objectForKey:@(dataTask.taskIdentifier)];
+    if (!responseObj) {
+        NSMutableData * responseData = [NSMutableData dataWithData:data];
+        NSInteger statusCode = [(NSHTTPURLResponse *)[dataTask response] statusCode];
+        responseObj = [NSMutableDictionary dictionaryWithObjectsAndKeys: @(statusCode), @"statusCode",responseData, @"responseData", nil];
+        [uploadTaskResponses setValue:responseObj forKey:(NSString*)@(dataTask.taskIdentifier)];
+    } else {
+        [[responseObj objectForKey:@"responseData"] appendData:data ];
+    }
+}
 
-  if (session.configuration.identifier) {
-    [dict setObject:session.configuration.identifier forKey:@"sessionIdentifier"];
-  }
+-(void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
+{
+    //FunctionName();
+    
+    NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                                 [NSNumber numberWithUnsignedInteger:task.taskIdentifier], @"taskIdentifier",
+                                 nil];
+    
+    if (session.configuration.identifier) {
+        [dict setObject:session.configuration.identifier forKey:@"sessionIdentifier"];
+    }
+    
+    if (error) {
+        NSDictionary * errorinfo = [NSDictionary dictionaryWithObjectsAndKeys:NUMBOOL(NO), @"success",
+                                    NUMINTEGER([error code]), @"errorCode",
+                                    [error localizedDescription], @"message",
+                                    nil];
+        [dict addEntriesFromDictionary:errorinfo];
+    } else {
+        NSMutableDictionary *responseObj = [uploadTaskResponses objectForKey:@(task.taskIdentifier)];
```

Comment:
Again, why mutable here? It does not mutate

Reply:
Actually there is another serious issue. I was reviewing this code today and found I was doing this
```
NSInteger *  statusCode = nil;
```
I've changed it to this
```
NSInteger  statusCode = 0;
```
For some reason this code was working allright , but today I tested for the first time with an iPhone 5C and it was causing a bad access exception when adding to the dictionary.

Reply:
well, you are basically making a pointer-object out of the underlaying primitive type. But assigning a primitive to the pointer then likely crashes it. Let me know when you fixed the other issues, then I do an additional commit to resolve other related issues.